cmd/tap/README.md

```diff
@@ -0,0 +1,179 @@
+`tap`: atproto sync utility
+========================================
+
+Tap is a single-tenant service that subscribes to an atproto relay and outputs filtered, verified events for a subset of repos.
+
+Tap simplifies firehose consumption by handling verification, backfill, and filtering. Your application connects to tap and receives simple JSON events for only the repos and collections you care about. Historical data for configured repos is automatically fetched from PDSs and delivered before live events begin.
+
+Features and design decisions:
+
+- verifies repo structure, MST integrity, and identity signatures
+- automatic backfill: fetches full repo history from PDS when adding new repos
+- filtered output: by DID list, by collection, or full network mode
+- ordering guarantees: live events wait for historical backfill to complete
+- delivery modes: WebSocket with acks, fire-and-forget, or webhook
+- single golang binary, SQLite backend
+- designed for moderate scale (thousands of repos, 10k+ events/sec)
+
+This tool is useful for building applications that need to track specific accounts or collections without dealing with the complexity of repo verification and backfill orchestration.
+
+⚠️ This tool is still in beta and may have bugs. Please report any issues you encounter.
+
+## Quick Start
+
+```bash
+# Run tap
+go run ./cmd/tap --disable-acks=true
+# By default, the service uses SQLite at `./tap.db` and binds to port `:8080`.
+
+# In a separate terminal, connect to receive events:
+websocat ws://localhost:8080/channel
+
+#  Add a repo to track
+curl -X POST http://localhost:8080/add-repos \
+  -H "Content-Type: application/json" \
+  -d '{"dids": ["did:plc:z72i7hdynmk6r22z27h6tvur"]}' # @bsky.app repo
+```
+
+Each repo will be backfilled from its PDS, then live events will stream as they arrive from the relay.
+
+## HTTP API
+
+- `GET /health`: returns `{"status":"ok"}`
+- `POST /add-repos`: add DIDs to track (triggers backfill of added repos)
+- `POST /remove-repos`: remove DIDs (stops sync, deletes data. does not delete buffered events in outbox)
+- `GET /channel`: WebSocket endpoint to receive events
+
+If more than one client connects to the WebSocket, events will be transparently sharded across all connected clients.
```

Comment:
are there any guarantees about this sharding? eg, will the same all messages for the same DID go to the same connection? gets weird with re-connections I suppose.